### PR TITLE
Partial #55 - splitting the worker out

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

File: aimmo-game/simulation/world_map.py [hunk 1]
```diff
@@ -39,6 +42,15 @@ def __repr__(self):
     def __eq__(self, other):
         return self.location == other.location
 
+    def serialise(self):
+        return {
+            'avatar': self.avatar.serialise() if self.avatar else None,
+            'generates_score': self.generates_score,
+            'habitable': self.habitable,
+            'location': self.location.serialise(),
```
Comment (jishi9) [inline aimmo-game/simulation/world_map.py]: no None-check needed here too?

File: aimmo-game-worker/simulation/direction.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# TODO: investigate using x and y
```
Comment (jishi9) [inline aimmo-game-worker/simulation/direction.py]: This is already using x and y ?

File: aimmo-game-worker/test-initialise.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
```
Comment (jishi9) [inline aimmo-game-worker/test-initialise.py]: what's this file for out of interest?

File: aimmo-game/simulation/turn_manager.py [hunk 4]
```diff
@@ -33,13 +35,15 @@ def set_world(self, new_world_state):
 world_state_provider = WorldStateProvider()
 
 
-class TurnManager(object):
+class TurnManager(threading.Thread):
     """
     Game loop
     """
+    daemon = True
```
Comment (jishi9) [inline aimmo-game/simulation/turn_manager.py]: unused

Comment (jishi9) [inline aimmo-game/simulation/turn_manager.py]: not sure, but should this be set in `self.daemon` ?

File: aimmo-game/simulation/worker_manager.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import logging
+import threading
+import time
+
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+class WorkerManager(threading.Thread):
+    daemon = True
```
Comment (jishi9) [inline aimmo-game/simulation/worker_manager.py]: not sure, but should this be set in `self.daemon` ?

File: aimmo-game/simulation/turn_manager.py [hunk 6]
```diff
@@ -49,27 +53,26 @@ def run_turn(self):
...
+                try:
+                    data = requests.post(avatar.worker_url, json=turn_state).json()
+                except ValueError as err:
+                    LOGGER.info("Failed to get turn result: %s", err)
+                else:
+                    try:
+                        action_data = data['action']
+                        action = ACTIONS[action_data['action_type']](**action_data.get('options', {}))
+                    except (KeyError, ValueError) as err:
+                        LOGGER.info("Bad action data supplied: %s", err)
```
Comment (jishi9) [inline aimmo-game/simulation/turn_manager.py]: `action.apply` is called after this, rather than re-raising the exception

**Comment by jishi9** [comment]
Tested locally, seems to work fine.

File: aimmo-game/simulation/world_map.py (hunk 1 above)
Comment (mikebryant) [inline aimmo-game/simulation/world_map.py]: Every cell has a location! Or it should...

File: aimmo-game-worker/simulation/direction.py (hunk 2 above)
Comment (mikebryant) [inline aimmo-game-worker/simulation/direction.py]: I think that may have been left over..

File: aimmo-game/simulation/worker_manager.py (hunk 5 above)
Comment (mikebryant) [inline aimmo-game/simulation/worker_manager.py]: Nope, you can set it on the class level and it'll work just as well

File: aimmo-game/simulation/avatar/avatar_wrapper.py [hunk 7]
```diff
@@ -57,14 +22,18 @@ def die(self, respawn_location):
...
-            try:
-                exec(code)
-                self.avatar = Avatar()
-            except Exception as ex:
-                raise UserCodeException("Exception in user code", ex)
+    def serialise(self):
+        return {
+            'events': [
+            #    {
+            #        'event_name': event.__class__.__name__.lower(),
```
Comment (jishi9) [inline aimmo-game/simulation/avatar/avatar_wrapper.py]: will this be revived in a similar form later, or should we just delete it?

File: aimmo-game/simulation/turn_manager.py (hunk 6 above)
Comment (mikebryant) [inline aimmo-game/simulation/turn_manager.py]: Yeh, this should be in an else: clause, I guess

File: aimmo-game-worker/test-initialise.py (hunk 3 above)
Comment (mikebryant) [inline aimmo-game-worker/test-initialise.py]: This was being used after development of the game worker, before the game itself starting sending requests, for basic testing

File: aimmo-game/simulation/avatar/avatar_wrapper.py (hunk 7 above)
Comment (mikebryant) [inline aimmo-game/simulation/avatar/avatar_wrapper.py]: I'm not sure what's happening with events right now. I think we probably will want them, in some form or another

File: aimmo-game/simulation/turn_manager.py (hunk 4 above)
Comment (mikebryant) [inline aimmo-game/simulation/turn_manager.py]: Nope, it's fine being in the class

**Pull request merged**